web/src/settings_account.ts

```diff
@@ -832,8 +833,15 @@ export function set_up(): void {
     $<HTMLSelectElement>("select#user_timezone").on("change", function (e) {
         e.preventDefault();
         e.stopPropagation();
-
-        const data = {timezone: this.value};
+        const chosen_timezone = timerender.browser_canonicalize_timezone(this.value);
+        const data = {
+            timezone: chosen_timezone,
+            web_suggest_update_timezone: chosen_timezone === timerender.display_time_zone,
+        };
+        $("#automatically_offer_update_time_zone").prop(
+            "checked",
+            data.web_suggest_update_timezone,
+        );
```

Comment:
Do we need to check it here? I'd expect that the `server_events_dispatch` handler for the server executing the settings change would update the UI. If not, that's a bug we need to investigate.

Reply:
Sure, I've shifted it to `server_events_dispatch`.